geode-core/src/distributedTest/java/org/apache/geode/internal/cache/execute/FunctionServiceBase.java

```diff
@@ -433,4 +437,17 @@ public void fromData(DataInput in) throws IOException, ClassNotFoundException {
     }
   }
 
+  public static <T> Function<T> identifiableFunction(SerializableConsumerIF<FunctionContext<T>> f) {
+    return new Function<T>() {
+      @Override
+      public String getId() {
```

Comment:
Would it be better to fix this in the product by changing the default Function.getId method to still call getCanonicalName but then check if it returns null and in that case return class.getName()?
Otherwise users may hit the same issue with the product's default getId implementation.

Reply:
Is it okay to change behavior in a public API? If so, that would definitely be better.

Reply:
Something I find confusing about this just being an issue on jdk17 is that getCanonicalName on jdk8 says it will return null for an anonymous class:
Returns null if the underlying class does not have a canonical name (i.e., if it is a local or anonymous class or an array whose component type does not have a canonical name).

Reply:
Yes. The same problem happens with anonymous classes on _all_ JDKs. It happens with lambdas starting with JDK14 or 15. Before that, lambdas' classes had canonical names. Anonymous classes didn't.